crates/red_knot/src/db.rs

```diff
@@ -57,6 +57,7 @@ pub trait ParallelDatabase: Database + Send {
     /// We should avoid creating snapshots while running a query because we might want to adopt Salsa in the future (if we can figure out persistent caching).
     /// Unfortunately, the infrastructure doesn't provide an automated way of knowing when a query is run, that's
     /// why we have to "enforce" this constraint manually.
+    #[must_use]
```

Comment:
Clippy's [`must_use`](https://rust-lang.github.io/rust-clippy/master/index.html#/must_use_candidate) rule was yelling at me ;) but I think it's actually right.

You can mark struct/enums and methods as `must_use`. The compiler will then emit a warning if you call a function that returns a *must use* object that you aren't handling. You've probably seen these warnings before when calling a function that returns a `Result` that you aren't handling. The compiler generates these warnings because `Result` is marked as `must_use`. 

I mark the `snapshot` function here as `must_use` because calling the function is useless if you aren't using the result. That means, it's most likely an error and you wanted to assign the result to a variable.

Reply:
Ah, ok, that makes perfect sense. I understand what `#[must_use]` does, I was just trying to figure out if there was some correctness or safety invariant that would be violated by failure to use the return value of this. But even if not it makes sense that it would likely be a bug not to use it, and it's nice to flag that likely bug.